master/internal/stream/system.go

```diff
@@ -210,46 +297,46 @@ func (ps PublisherSet) Websocket(socket *websocket.Conn, c echo.Context) error {
 
 	for {
 		mods, msgs, closed := waitForSomething()
-
 		// were we closed?
 		if closed {
 			return nil
 		}
 
 		// any modifications to our subscriptions?
 		for _, mod := range mods {
-			temp, err := ss.SubscriptionMod(mod, ctx)
+			temp, err := ss.SubscriptionMod(ctx, mod)
 			if err != nil {
 				return errors.Wrapf(err, "error modifying subscriptions")
 			}
 			msgs = append(msgs, temp...)
-			// TODO: also append a sync message (or one sync per SubscriptionModMsg)
+			// XXX: also append a sync message (or one sync per SubscriptionModMsg)
 		}
 
 		// write msgs to the websocket
 		err = writeAll(socket, msgs)
 		if err != nil {
-			// TODO: don't log broken pipe errors.
+			// XXX: don't log broken pipe errors.
 			if err != nil {
 				return errors.Wrapf(err, "error writing to socket")
 			}
 		}
 	}
-
-	return nil
 }
 
+// publishLoop monitors for new events and broadcasts them to Publishers.
 func publishLoop[T stream.Msg](
 	ctx context.Context,
 	channelName string,
 	publisher *stream.Publisher[T],
 ) {
-	// TODO: is there a better recovery technique than this?
-	// XXX: at least boot all the connected streamers, they'll all be invalid now
+	// XXX: is there a better recovery technique than this?
```

Comment:
yes!  Use the new supervisor you wrote!

Reply:
yep! I'll pass up the errors here and have the supervisor due it job at recovering from errors during `Run()`